src/include/function/gds/gds_utils.h

```diff
@@ -9,9 +9,12 @@ namespace function {
 
 class KUZU_API GDSUtils {
 public:
-    static void runFrontiersUntilConvergence(processor::ExecutionContext* context,
-        GDSComputeState& rjCompState, graph::Graph* graph, common::ExtendDirection extendDirection,
-        uint64_t maxIteration, const std::string& propertyToScan = "");
+    static void runGDS(processor::ExecutionContext* context, GDSComputeState& compState,
+        graph::Graph* graph, common::ExtendDirection extendDirection, uint64_t maxIteration);
+    static void runRecursiveJoin(processor::ExecutionContext* context, GDSComputeState& compState,
```

Comment:
It looks a bit weird that we have a separate `runRecursiveJoin` under `GDSUtils`. Do you want to separate this into another Utils class?